### PR TITLE
✨ added retries for upload and download in the node ports package

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

This PR adds retries in case of ConnectionError (see [aiohttp docs](https://docs.aiohttp.org/en/stable/client_reference.html#connection-errors))

It will retry using a default of 5 minutes, changeable via ENV variable

## Related issue/s
fixes https://github.com/ITISFoundation/osparc-simcore/issues/3242

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]
Nice. some feedback below!

**Review by odeimaiz — APPROVED** [review]
👌 👌

File: swagger-cli [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules/swagger-cli/swagger-cli.js
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply swagger-cli:1]: oups that is a unwanted commit. will revert that.

File: packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py [hunk 2]
```diff
@@ -202,28 +219,38 @@ async def _upload_file_part(
...
+                upload_url,
+                data=file_uploader,
+                headers={
+                    "Content-Length": f"{file_part_size}",
+                },
+            )
+            response.raise_for_status()
+            pbar.update(file_part_size)
+            # NOTE: the response from minio does not contain a json body
+            assert response.status == web.HTTPOk.status_code
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py:242]: I don't know. I was actually also wondering

File: packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py [hunk 3]
```diff
@@ -127,27 +134,37 @@ async def _get_upload_links(
...
-                        await file_pointer.write(chunk)
-                        pbar.update(len(chunk))
-                log.debug("Download complete")
-        except ClientPayloadError as exc:
-            raise exceptions.TransferError(url) from exc
+    async for attempt in AsyncRetrying(
+        reraise=True,
+        wait=wait_random(min=0.1, max=2),
+        stop=stop_after_delay(
+            NodePortsSettings.create_from_envs().NODE_PORTS_IO_RETRY_DELAY_S
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py:141]: I am not sure I get what you mean by that comment?

File: packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py [hunk 4]
```diff
@@ -127,27 +134,37 @@ async def _get_upload_links(
...
-                async with aiofiles.open(file_path, "wb") as file_pointer:
-                    while chunk := await response.content.read(CHUNK_SIZE):
-                        await file_pointer.write(chunk)
-                        pbar.update(len(chunk))
-                log.debug("Download complete")
-        except ClientPayloadError as exc:
-            raise exceptions.TransferError(url) from exc
+    async for attempt in AsyncRetrying(
+        reraise=True,
+        wait=wait_random(min=0.1, max=2),
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_common/filemanager.py:139]: well ok. and then with exponential what parameters?
I see this: [https://tenacity.readthedocs.io/en/latest/#waiting-before-retrying](https://tenacity.readthedocs.io/en/latest/#waiting-before-retrying)

I could also go with fixed+random... I mean we don't have anything defined.

**Pull request merged**